public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -359,32 +363,50 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       (event: SceneObjectStateChangedEvent) => {
         if (event.payload.changedObject instanceof SceneRefreshPicker) {
           if (Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'intervals')) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
         if (event.payload.changedObject instanceof SceneGridItem) {
-          this.setIsDirty();
+          this.detectChanges();
         }
         if (event.payload.changedObject instanceof DashboardScene) {
           if (Object.keys(event.payload.partialUpdate).some((key) => PERSISTED_PROPS.includes(key))) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
         if (event.payload.changedObject instanceof SceneTimeRange) {
-          this.setIsDirty();
+          this.detectChanges();
         }
         if (event.payload.changedObject instanceof DashboardControls) {
           if (Object.prototype.hasOwnProperty.call(event.payload.partialUpdate, 'hideTimeControls')) {
-            this.setIsDirty();
+            this.detectChanges();
           }
         }
+        if (event.payload.changedObject instanceof SceneVariableSet) {
+          this.detectChanges();
+        }
+        if (event.payload.changedObject instanceof DashboardAnnotationsDataLayer) {
+          this.detectChanges();
```

Comment:
I think we need to also add a check for the `query` property here. More details in this [discussion](https://github.com/grafana/grafana/pull/81361#discussion_r1481247728)